src/gluonts/dataset/split.py

```diff
@@ -372,20 +372,39 @@ def __iter__(self) -> Generator[Tuple[DataEntry, DataEntry], None, None]:
             max_history=self.max_history,
         )
 
+    def __len__(self):
+        return len(self.dataset) * self.windows
+
     @property
-    def input(self) -> Generator[DataEntry, None, None]:
-        """
-        Iterable over the ``input`` portion of the test data.
-        """
-        for input, _ in self:
-            yield input
+    def input(self) -> "InputDataset":
+        return InputDataset(self)
 
     @property
-    def label(self) -> Generator[DataEntry, None, None]:
-        """
-        Iterable over the ``label`` portion of the test data.
-        """
-        for _, label in self:
+    def label(self) -> "LabelDataset":
+        return LabelDataset(self)
+
+
+@dataclass
+class InputDataset:
+    test_data: TestData
+
+    def __len__(self):
+        return len(self.test_data)
+
+    def __iter__(self):
+        for input, _ in self.test_data:
```

Comment:
I guess there are so many other ways to do this, but this should be fine.

However, can we maybe also deconstruct the second argument to clarify what we are discarding?